intermediate_source/parametrizations.py

```diff
@@ -0,0 +1,353 @@
+# -*- coding: utf-8 -*-
+"""
+Parametrizations Tutorial
+=========================
+**Author**: `Mario Lezcano <https://github.com/lezcano>`_
+
+Regularizing deep-learning models is a surprisingly challenging task.
+Classical techniques such as penalty methods often fall short when applied
+on deep models due to the complexity of the function being optimized.
+This is particularly problematic when working with ill-conditioned models.
+Examples of these are RNNs trained on long sequences and GANs. A number
+of techniques have been proposed in the recent years to regularize these
+models and improve their convergence. On recurrent models, it has been
+proposed to control the singular values of the recurrent kernel for the
+RNN to be well-conditioned. This can be achieved, for example, by making
+the recurrent kernel `orthogonal <https://en.wikipedia.org/wiki/Orthogonal_matrix>`_.
+Another way to regularize recurrent models is via
+"`weight normalization <https://pytorch.org/docs/stable/generated/torch.nn.utils.weight_norm.html>`_".
+This approach proposes to decouple the learning of the parameters from the
+learning of their scale.  To do so, the parameter is divided by its
+`Frobenius norm <https://en.wikipedia.org/wiki/Matrix_norm#Frobenius_norm>`_.
+A similar regularization was proposed for GANs under the name of
+"`spectral normalization <https://pytorch.org/docs/stable/generated/torch.nn.utils.spectral_norm.html>`_". This method
+controls the Lipschitz constant of the network by dividing its parameters by
+their `spectral norm <https://en.wikipedia.org/wiki/Matrix_norm#Special_cases>`_,
+rather than its Frobenius norm.
+
+All these methods have a pattern in common. They all transform a parameter
+in an appropriate way before using it. In the first case, they make it orthogonal by
+using a function that maps matrices to orthogonal matrices. In the case of weight
+and spectral normalization, they divide the original parameter by its norm.
+
+More generally, all these examples use a function to put extra structure on the parameters.
+In other words, they use a function to constrain the parameters.
+
+In this tutorial, you will learn how to implement and use this pattern to write and
+put constraints on your model. Doing so is as easy as writing your own ``nn.Module``.
+
+Requirements
+------------
+``"torch>=1.9.0a0+7aeee28"``
```

Comment:
I think that 1.9.0 would be enough here no?
I don't think users know exactly what the hash means and how to check if their nightly version is before or after this one hash.